main/src/main/scala/sbt/internal/RemoteCache.scala

```diff
@@ -162,7 +162,10 @@ object RemoteCache {
           val original = packageBin.in(Defaults.TaskZero).value
           val artp = artifactPath.value
           val af = compileAnalysisFile.value
-          IO.copyFile(original, artp)
+          // when a file is copied to itself, it becomes zero size
+          if (!original.equals(artp)) {
```

Comment:
So this means `Compile/packageBin` and `Compile/packageCache` would end up manipulating the same JAR file? I don't know if that's safe either.

Reply:
I have not yet figured out why they are the same.

Reply:
Perhaps then an exception indicating the settings affecting these two files will be useful.